Sources/ElasticSwift/Responses/Response.swift

```diff
@@ -21,20 +22,61 @@ public class ESResponse {
     }
 }
 
-public class GetResponse<T: Codable>: Codable {
-    
-    public var index: String?
-    public var type: String?
-    public var id: String?
-    public var version: Int?
-    public var found: Bool?
+struct ResponseConstants {
+    struct Errors {
+        enum ResponseError : Error {
+            case NoDataReturned
+        }
+    }
+}
+
+public protocol Response : Codable {
     
-    public var source: T?
+    static func create(fromESResponse response: ESResponse, withSerializer serializer: Serializer) throws -> Self
+}
+
+extension Response {
```

Comment:
Common place to manage responses and deserialization.

See previous comment about managing responses